docs/react-v9/contributing/patterns/change-event-callbacks.md

```diff
@@ -0,0 +1,123 @@
+## Change Event Callbacks
+
+Fluent UI React V9 implements event callbacks to expose internal state changes to consumers. These callbacks pass both the event data and the underlying React or DOM event for consumption.
+
+An example of this is the boolean change on Dialog to open or close via internal interactions, surfacing these events ensures that controlled implementations can update based on internal state.
+
+```
+// Dialog.types..ts
+  export type DialogOpenChangeEvent = DialogOpenChangeData['event'];
+
+  export type DialogOpenChangeData =
+  | {
+      type: 'escapeKeyDown';
+      open: boolean;
+      event: React.KeyboardEvent<DialogSurfaceElement>;
+    }
+  | {
+      type: 'backdropClick';
+      open: boolean;
+      event: React.MouseEvent<DialogSurfaceElement>;
+    }
+  | {
+      type: 'triggerClick';
+      open: boolean;
+      event: React.MouseEvent<DialogSurfaceElement>;
+  };
+  /**
+  * Callback fired when the component changes value from open state.
+  *
+  * @param event - a React's Synthetic event or a KeyboardEvent in case of `documentEscapeKeyDown`
+  * @param data - A data object with relevant information,
+  * such as open value and type of interaction that created the event
+  */
+  export type DialogOpenChangeEventHandler = (event: DialogOpenChangeEvent, data: DialogOpenChangeData) => void;
+
+  /**
+   * Callback fired when the component changes value from open state.
+   *
+   * @param event - a React's Synthetic event or a KeyboardEvent in case of `documentEscapeKeyDown`
+   * @param data - A data object with relevant information,
+   * such as open value and type of interaction that created the event
+   */
+  // eslint-disable-next-line @nx/workspace-consistent-callback-type -- can't change type of existing callback
```

Comment:
This is a deprecated callback type. What about use an example that uses the new callback type? For example TagPicker `onOpenChange`? https://github.com/microsoft/fluentui/blob/d77384f0183852aeb17d3e3fccc9543be041df42/packages/react-components/react-tag-picker/library/src/components/TagPicker/TagPicker.types.ts#L43

Reply:
I see the first example is all taken from dialog. So maybe just add a comment somewhere here pointing to a component that uses the new type like TagPicker

Reply:
Good catch thank you! I'll replace this with the latest